pyprocessing/math.py

```diff
@@ -162,191 +179,214 @@ def from_angle(angle, target=None):
-        )
+        if not isinstance(other, Iterable):
+            return False
+
+        if not len(other) == len(self):
+            return False
+
+        if not all(component_a == component_b for
+                   component_a, component_b in zip(self, other)):
+            return False
+
+        return True
 
     def __getitem__(self, item):
-        if isinstance(item, str):
-            return {'x': self.x, 'y': self.y, 'z': self.z}[item]
-        if isinstance(item, int):
-            return [self.x, self.y, self.z][item]
-        raise KeyError(f"Item '{item}' cannot be accessed.")
+        return self.seq[item]
+
+    def __setitem__(self, item, v):
+        self.seq[item] = v
+
+    @property
+    def position(self):
+        return tuple(self)
 
     @property
-    def z(self):
-        if self._z is None:
-            return 0
-        return self._z
+    def is_2d(self):
+        return len(self) == 2
 
-    @z.setter
-    def z(self, value):
-        self._z = value
-        if value is not None:
-            self.is_2d = False
+    @property
+    def is_3d(self):
+        return len(self) == 3
 
     @property
-    def position(self):
-        return self.x, self.y, self.z
+    def is_4d(self):
+        return len(self) == 4
 
     def set(self, *vector, x=None, y=None, z=None):
-        if vector:
-            if isinstance(vector[0], PVector):
-                self.x, self.y, self.z = vector[0].position
-            elif isinstance(vector[0], (list, tuple)):
-                self.x, self.y, *_ = vector[0]
-                if len(vector[0]) == 3:
-                    self.z = vector[0][2]
-                else:
-                    self.z = None
-            elif all(isinstance(p, (int, float)) for p in vector):
-                self.x, self.y, self.z = vector
-            elif isinstance(vector[0], complex):
-                self.x, self.y, self.z = vector[0].real, vector[0].imag, None
-        if x is not None:
-            self.x = x
-        if y is not None:
-            self.y = y
-        if z is not None:
-            self.z = z
+        self.seq = PVector(*vector, x, y, z).seq
 
     def copy(self):
-        return PVector(self.x, self.y, self._z)
+        return PVector(*self)
 
     def mag(self):
         return math.sqrt(self.mag_sq())
 
     def mag_sq(self):
-        return self.x * self.x + self.y * self.y + self.z * self.z
+        return sum(component_a * component_a for component_a in self)
 
     def dot(self, other):
-        return self.x * other.x + self.y * other.y + self.z * other.z
+        return sum(component_a * component_b for component_a, component_b in zip(self, other))
 
     def cross(self, other):
-        x = self.y * other.z - self.z * other.y
-        y = self.z * other.x - self.x * other.z
-        z = self.x * other.y - self.y * other.x
+        if not len(self) == 3:
+            raise ValueError(
+                'Cross product is only defined for 3d vectors'
```

Comment:
it's also defined in 7 dimensions by the way (it's just FYI). It may be a good idea to implement the Gram-Schmidt algorithm to find orthgonized bases in all dimensions

Reply:
Really?, I didn't know that, well, thats a little too advanced for my current knowledge of linear algebra.
I will just change the exception text for now since it will take me at least a month to learn it, if I can ever learn.